core/modules/replica/python/schema_migration.py

```diff
@@ -0,0 +1,219 @@
+"""Module defining methods used in schema migration of qservReplica database.
+"""
+
+__all__ = ["make_migration_manager"]
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+import logging
+import os
+import re
+
+#-----------------------------
+# Imports for other modules --
+#-----------------------------
+from lsst.qserv.schema import SchemaMigMgr
+from lsst.db import utils
+
+#----------------------------------
+# Local non-exported definitions --
+#----------------------------------
+
+_log = logging.getLogger(__name__)
+
+#------------------------
+# Exported definitions --
+#------------------------
+
+
+class QservReplicaMigrationManager(SchemaMigMgr):
+    """Class implementing schema migration for qservReplica database.
+    """
+
+    # migration script has name "migrate-N-to-M.sql" or "migrate-N-to-M-something.sql"
+    # when applying scripts for any given version N the names are sorted first.
+    _mig_name_re = re.compile(r"migrate-(?P<from>\d+)-to-(?P<to>\d+)(-.*)?.sql")
+
+    def __init__(self, name, engine, scripts_dir):
+
+        SchemaMigMgr.__init__(self)
+
+        self.engine = engine
+
+        # scripts are located in replica/ sub-dir
+        scripts_dir = os.path.join(scripts_dir, name)
+
+        # find all migration scripts, add full script path to each
+        scripts = self.findScripts(scripts_dir, self._script_match)
+        self.scripts = [(v0, v1, fname, os.path.join(scripts_dir, fname))
+                        for v0, v1, fname in scripts]
```

Comment:
Instead of 4-item tuples I'd consider writing a container class, or maybe at least use a `namedtuple`. Data structures like these tend toward entropy and can become dangerously complicated.